pkg/proxy/ipvs/proxier.go

```diff
@@ -1693,23 +1693,26 @@ func (proxier *Proxier) cleanLegacyService(activeServices map[string]bool, curre
 		if _, ok := activeServices[cs]; !ok {
 			rsList, _ := proxier.ipvs.GetRealServers(svc)
 
-			// If we still have real servers graceful termination is not done
-			if len(rsList) > 0 {
-				continue
-			}
-
+			okayToDeleteVS := true
 			// Applying graceful termination to all real servers
 			for _, rs := range rsList {
 				uniqueRS := GetUniqueRSName(svc, rs)
 				// If RS is already in the graceful termination list, no need to add it again
 				if proxier.gracefuldeleteManager.InTerminationList(uniqueRS) {
+					okayToDeleteVS = false
```

Comment:
Can you please share more (either as a comment or just responding to this code review) about why its not ok to delete VS if proxier.gracefuldeleteManager.InTerminationList(uniqueRS) is true?

Reply:
if proxier.gracefuldeleteManager.InTerminationList(uniqueRS) is true meant it have some RS exist not deleted,  only when the whole RS of VS is deleted, we should delete the VS done.